pkg/discovery/worker/aggregation/container_utilization_data_aggregator.go

```diff
@@ -0,0 +1,92 @@
+package aggregation
+
+import (
+	"fmt"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+	"math"
+)
+
+var (
+	maxUtilizationDataStrategy = "maxUtilizationData"
+	allUtilizationDataStrategy = "allUtilizationData"
+
+	DefaultContainerUtilizationDataAggStrategy = maxUtilizationDataStrategy
+
+	// Map from the configured utilization data aggregation strategy to utilization data aggregator
+	ContainerUtilizationDataAggregators = map[string]ContainerUtilizationDataAggregator{
+		maxUtilizationDataStrategy: &maxUtilizationDataAggregator{aggregationStrategy: "max utilization data strategy"},
+		allUtilizationDataStrategy: &allUtilizationDataAggregator{aggregationStrategy: "all utilization data strategy"},
+	}
+)
+
+// ContainerUtilizationDataAggregator interface represents a type of container utilization data aggregator
+type ContainerUtilizationDataAggregator interface {
+	// AggregationStrategy returns aggregation strategy of this data aggregator
+	AggregationStrategy() string
+	// Aggregate aggregates commodities utilization data based on the given list of commodity DTOs of a commodity type
+	// and aggregation strategy, and returns aggregated utilization data which contains utilization data points, last
+	// point timestamp milliseconds and interval milliseconds
+	Aggregate(commodities []*proto.CommodityDTO, lastPointTimestampMs int64) ([]float64, int64, int32, error)
+}
+
+// ---------------- All utilization data aggregation strategy ----------------
+type allUtilizationDataAggregator struct {
+	aggregationStrategy string
+}
+
+func (allDataAggregator *allUtilizationDataAggregator) AggregationStrategy() string {
+	return allDataAggregator.aggregationStrategy
+}
+
+func (allDataAggregator *allUtilizationDataAggregator) Aggregate(commodities []*proto.CommodityDTO,
+	lastPointTimestampMs int64) ([]float64, int64, int32, error) {
```

Comment:
lastPointTimestampMs ? Whats the `M`?. Also it does not seem to be used in either interface implementers. Is there more functionality coming?

Reply:
The "Ms" means milliseconds. Sorry for the confusion. I named "lastPointTimestampMs" based on the same name from CommodityDTO. For now it's always set to current unix milliseconds. Will remove this argument to be less confusing and keep the returned []float64 only.